pkg/minikube/kubeconfig/setup.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconfig
+
+import (
+	"io/ioutil"
+	"sync/atomic"
+
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+// Setup is the kubeconfig setup
```

Comment:
I feel like a struct with these sort of values would typically be called Config... but it's not, because it has functions on it. It doesn't quite feel right to me.